patricia/patricia.go

```diff
@@ -242,6 +261,11 @@ func (trie *Trie) Delete(key Prefix) (deleted bool) {
 		}
 	}
 
+	// Remove the node if it has no items.
+	if node.size() == 0 {
```

Comment:
@glyn Might be a bit more efficient to use something like

```go
func (trie *Trie) empty() bool {
    isEmpty := true

    trie.walk(nil, func(prefix Prefix, item Item) error {
        isEmpty = false
        return SkipSubtree
    })  

    return isEmpty
}
```

In this way you return as soon as you detect a node that is not empty. We don't really need to get the actual size for our use case and walk the whole Trie, right?